scripts/jenkins/cloud/ansible/roles/setup_zypper_repos/tasks/add_extra_repos.yml

```diff
@@ -15,9 +15,41 @@
 #
 ---
 
-- name: Add extra repos
-  zypper_repository:
-    repo: "{{ item }}"
+- name: Download packages from extra repositories
+  command: |
+    wget --progress=dot:mega -r --directory-prefix {{ local_repos_base_dir }}/PTF \
+      -e robots=off  --no-check-certificate --no-parent --no-clobber --accept \
+       x86_64.rpm,noarch.rpm {{ item }}
+  args:
+    warn: false
   loop: "{{ extra_repos_list }}"
+  register: rpm_download_status
+  until: rpm_download_status is succeeded
+  retries: 5
+  delay: 5
   notify:
     - Refresh all repos
+
+- block:
+    - name: Update PTF repo metadata
+      command: createrepo-cloud-ptf
+
+  rescue:
+    - name: Esnsure createrepo tool installed
+      zypper:
+        name: createrepo
+        disable_gpg_check: true
+
+    - name: Create PTF repository
+      command: createrepo -o {{ local_repos_base_dir }}/PTF {{ local_repos_base_dir }}/PTF
+
+  always:
+    - name: Ensure zypper PTF repository
+      zypper_repository:
+        name: "PTF"
+        priority: 90
```

Comment:
Why do we have higher prio only on Admin node? I know that crowbar has it in that way but is it how it should be?

This can create unexpected results in case that you put a repo with lower version than in update repo(for example). Think of revalidating a potential regression (custom repository in IBS). So on admin u have package from your repo but on the nodes there is a version from product channel. 

I expect that once I put extra repo then it will be used no matter what.. that is why it is "extra repo", right? :)

Reply:
After some discussion with @stefannica we have decided to not use the PTF repository to provide packages from `extra_repos`, as we would be affecting the product if we create and set the priority of the PTF repository which is something done by ardana itself.

The new implementation creates a new repository called `extra-repos` on the deployer node and all packages from the `extra_repos` will be downloaded and made available through this repository. This repository will be configured on all cloud nodes with higher priority.

Reply:
Ok